### ISSUE TITLE
Loading from GCS with put-file hangs even though completes successfully

### ISSUE THREAD

**Issue by rj3d** [pachyderm/pachyderm#1906]
The last couple times I've loaded from GCS with a multiline start-commit, put-file, and finish-commit the put-file successfully loads data (verified by file sizes), but instead hangs and doesn't exit. I have to Ctrl+C out and then run finish-commit again to close the commit. This isn't a big issue with small commits, but when loading large datasets it can get kind of nerve-wracking to guess when things are done.

An example command that has hung:
```
pachctl start-commit somatic_variant_aligner_input master && \
pachctl put-file somatic_variant_aligner_input master /tesla -r -f gs://tesla-data/raw-data -f gs://tesla-data/metadata/intervals && \
pachctl finish-commit somatic_variant_aligner_input master
```

I looked through the logs, and there weren't any error messages during the put-file. It looks like it completes successfully, but for whatever reason the put-file doesn't exit. Let me know if I can provide anything to help.

Thanks!

**Comment by derekchiang** [comment]
@rj3d could you upload the logs to pastebin/gist and post a link here?

Also, to be clear, was it `put-file` or `finish-commit` that was hanging?

**Comment by rj3d** [comment]
@derekchiang [Here are the logs](https://pastebin.com/nNzCJ3xV) from creating the repo to a little bit after when the load finished. I'm not sure if it was `put-file` or `finish-commit` that was hanging since I chained the commands, but I was assuming that it was `put-file` since finish-commit has run instantaneously and fine the first time every time after using Control+C to get out.

**Comment by derekchiang** [comment]
@rj3d are these logs from when the hang occurred?

**Comment by sjezewski** [comment]
Huh. Logs suggest that the putfile completed (~5000s or ~1hr 20m) but I don't see a log line for the finish commit ... actually I don't see a starting log line for finish commit. Which way did you run this ... w `putfile -c` or a separate `finish-commit` call?

**Comment by rj3d** [comment]
@derekchiang  @sjezewski The dataset I'm using is ~75 GB spread across 8 files. I get the hang when I run this with a separate `start-commit` and `finish-commit` call. The longest I've let this run before exiting with Control+C has been ~18 hours. I just found a workaround though, using `put-file -c` works, and completes in ~7 minutes.

**Comment by rj3d** [comment]
I'm now seeing this issue with much larger commits even when using put-file with -c and using export ADDRESS instead of port forward. The command just hangs and never returns, and we can workaround if we allow command to run long enough that the `PutFile` finishes in the logs, Control+C out, and then manually run `finish-commit`.

One commit contains 91 files with a total size of 640.48 GiB and a maximum file size of 42.55 GiB and the other commit contains 22 files with a total size of 226.88 GiB and a maximum file size of 29.58 GiB.

I've been debugging this with @sjezewski, and according to his reading of the logs we see the `StartCommit`s in the logs, the `PutFile`s complete in the logs, but we don't see the `FinishCommit` api call in the logs.

**Comment by sjezewski** [comment]
We saw something similar a week ago with the `Process()` api call (before the PPS refactor to use a master model).

Basically the `Process()` call completed on the worker, but the debug line we put right after the `Process()` call on pps never printed. Somehow the grpc cxn was lost. In each case ... the deferred log printed suggesting that the server (pachd or worker in each case) thinks it successfully wrote the response.

I suspect an underlying grpc issue here. Somehow on long running requests we see a server send a response, but the client never receive it.

**Comment by JoeyZwicker** [comment]
I think this has been fixed by #2089. If i'm incorrect @derekchiang or @jdoliner, please reopne
